src/app/api/collections/route.ts

```diff
@@ -0,0 +1,26 @@
+import { NextRequest, NextResponse } from "next/server"
+import { initialize as initializeProductModule } from "@medusajs/product"
+import { notFound } from "next/navigation"
+
+export async function GET(request: NextRequest) {
+  const productService = await initializeProductModule()
+
+  const { offset } = Object.fromEntries(request.nextUrl.searchParams)
+
+  const [collections, count] = await productService.listAndCountCollections(
+    {},
+    {
+      skip: parseInt(offset) || 0,
+      take: 100,
+    }
+  )
+
+  if (!collections) {
```

Comment:
thought: Empty list responses typically don't respond with a 404, but not sure how this `notFound` function works

Reply:
yep it returns a 404. Agree it's a bit lazy, I'm just assuming that the requested collection handle doesn't exist. I'll try to handle this a bit more elegant.